src/data/data1d/filter1d/fft.js

```diff
@@ -13,23 +16,18 @@ export function apply(datum1D) {
     throw new Error('fft not applicable on this data');
   }
 
-  let re = new Float64Array(datum1D.data.re);
-  let im = new Float64Array(datum1D.data.im);
+  let digitalFilterApplied = datum1D.filters.some(
+    (e) => e.name === 'digitalFilter' && e.flag,
+  );
 
-  const nbPoints = re.length;
-  FFT.init(nbPoints);
+  Object.assign(datum1D.data, reimFFT(datum1D.data, { applyZeroShift: true }));
```

Comment:
Why not using

datum1D.data = reimFFT(datum1D.data, { applyZeroShift: true }

We need to keep the reference to the data property ?

Reply:
because reimFFT return { re, im } and data has x, re, im properties